cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java

```diff
@@ -57,14 +58,19 @@ public FileMetadataCleaner(TransactionRunner transactionRunner) {
     final List<DeletedEntry> toDelete = new ArrayList<>(fileCleanupBatchSize);
 
     try {
-      boolean reachedEnd = false;
+      final AtomicBoolean reachedEnd = new AtomicBoolean(false);
       final AtomicReference<Range> range = new AtomicReference<>(Range.all());
       // stop cleaning up if there are no files to delete or if we have reached batch size
-      while (!reachedEnd && toDelete.size() < fileCleanupBatchSize) {
+      while (!reachedEnd.get() && toDelete.size() < fileCleanupBatchSize) {
         TransactionRunners.run(transactionRunner, context -> {
           StructuredTable table = context.getTable(StoreDefinition.LogFileMetaStore.LOG_FILE_META);
-          try (CloseableIterator<StructuredRow> iter = table.scan(range.get(), fileCleanupBatchSize)) {
-            while (iter.hasNext() && toDelete.size() < fileCleanupBatchSize) {
+            try (CloseableIterator<StructuredRow> iter = table.scan(range.get(), fileCleanupBatchSize)) {
+            while (toDelete.size() < fileCleanupBatchSize) {
```

Comment:
If this loop is refactored to a separate method, then the logic becomes much simpler

Reply:
yes, that made code more readable.